cfme/middleware/__init__.py

```diff
@@ -53,3 +68,63 @@ def download(extension):
         download_btn("Download as {}".format(extensions_mapping[extension]))
     except:
         raise ValueError("Unknown extention. check the extentions_mapping")
+
+
+def add_deployment(filename, runtime_name=None, enable_deploy=True, cancel=False):
```

Comment:
I thought about it, but it can make mess because we will need 2 mixins.
1 Mixin can be extended by "Deployment", and it will contain deployment start/stop... already deployed objects operations
"add_deployment" method does not belong to "Deployment" and for that we will need second mixin for Server class.

Solution I gave was like cfme.web_ui.mixins but living in cfme/middleware/__init__.py